extra/utils.py

```diff
@@ -25,7 +26,7 @@ def download_file(url, fp, skip_if_exists=False):
     for chunk in r.iter_content(chunk_size=16384):
       progress_bar.update(f.write(chunk))
     f.close()
-    os.rename(f.name, fp)
+    shutil.move(f.name, fp)
```

Comment:
Is this guaranteed to be atomic?

Reply:
If they are on the same filesystem, then yes it just calls os.rename according to the docs. Otherwise it will first copy and then delete the old file if on different disks (no atomic across disks I believe). https://docs.python.org/3/library/shutil.html#shutil.move

Reply:
I think it should fail. The idea of this is to be atomic, and if it can't be, it should fail. The tempfile should be chosen to be on the same volume.

Reply:
I don't think there is a concept of a per-volume temp folder. If you want it to be atomic, it would probably be better to have this create a "temp" folder in the root of the repo and store everything in there instead of the system temp folder?